tests/Integration/UseCases/HandleCreditCardPaymentNotification/HandleCreditCardPaymentNotificationUseCaseTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Tests\Integration\UseCases\HandleCreditCardPaymentNotification;
+
+use Psr\Log\NullLogger;
+use WMDE\Fundraising\Frontend\DataAccess\DoctrineDonationRepository;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationConfirmationMailer;
+use WMDE\Fundraising\Frontend\Infrastructure\DonationEventLogger;
+use WMDE\Fundraising\Frontend\Tests\Data\ValidCreditCardNotificationRequest;
+use WMDE\Fundraising\Frontend\Tests\Data\ValidDonation;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\DonationRepositorySpy;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FailingDonationAuthorizer;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\FakeDonationRepository;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\SucceedingDonationAuthorizer;
+use WMDE\Fundraising\Frontend\Tests\Fixtures\ThrowingEntityManager;
+use WMDE\Fundraising\Frontend\UseCases\HandleCreditCardPaymentNotification\HandleCreditCardPaymentNotificationUseCase;
+
+/**
+ * @covers WMDE\Fundraising\Frontend\UseCases\HandleCreditCardPaymentNotification\HandleCreditCardPaymentNotificationUseCase
+ *
+ * @licence GNU GPL v2+
+ * @author Kai Nissen < kai.nissen@wikimedia.de >
+ */
+class HandleCreditCardPaymentNotificationUseCaseTest extends \PHPUnit_Framework_TestCase {
+
+	public function testWhenRepositoryThrowsException_handlerReturnsFalse() {
+		$useCase = new HandleCreditCardPaymentNotificationUseCase(
+			new DoctrineDonationRepository( ThrowingEntityManager::newInstance( $this ) ),
+			new FailingDonationAuthorizer(),
+			$this->getMailer(),
+			new NullLogger(),
+			$this->getEventLogger()
+		);
```

Comment:
I recommend following the pattern used here: https://github.com/wmde/FundraisingFrontend/blob/master/tests/Unit/UseCases/ApplyForMembership/MembershipApplicationValidatorTest.php#L27

* Have a setUp that initializes all services, each having a field
* Each service has minimal behaviour for the "success case"
* Individual test methods call a newWhateverThing method and don't know how the class under test is constructed
* Before calling newWhateverThing they set services as needed for the behaviour being tested (almost always a single one)